src/error.rs

```diff
@@ -7,123 +8,76 @@ use std::{
     io,
     path::PathBuf,
 };
-
-/// Wrapper around `serde_yaml::from_str`
+/// Shorthand for a Result that returns an `OrcaError`.
+pub type OrcaResult<T> = Result<T, OrcaError>;
```

Comment:
I'd actually prefer we just use `Result` and `Error` instead of `OrcaResult` and `OrcaError` following other crates' practices and also because I also do find `orcapod:error:Error` is enough of a specifier

Reply:
I also see this indirect `Error` struct to error Enum relationship found in library like `tokio` (https://docs.rs/tokio/latest/src/tokio/time/error.rs.html#26). This makes me believe that there is some decent reason to have this separation.

Reply:
Actually, I do see that more modern recommendation is to give more specific name like you did. Do you happen to have a good reference on what's a good practices on error implementation these days?